hack/tools/tilt-prepare/main.go

```diff
@@ -570,6 +582,87 @@ func certManagerTask() taskFunction {
 	}
 }
 
+// kustomizationFile represents a kustomization.yaml.
+type kustomizationFile struct {
+	HelmCharts []kustomizationFileHelmCharts `json:"helmCharts"`
+}
+type kustomizationFileHelmCharts struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
+}
+
+// chartFile represents a Chart.yaml.
+type chartFile struct {
+	Version string `json:"version"`
+}
+
+// cleanupChartTask cleans up outdated charts below the given path if they don't
+// match the pinned version. This is necessary because kustomize just uses a local
+// Chart if it exists.
+func cleanupChartTask(path string) taskFunction {
+	return func(ctx context.Context, prefix string, errCh chan error) {
+		err := filepath.WalkDir(path, func(path string, d fs.DirEntry, _ error) error {
+			if !strings.HasSuffix(path, "kustomization.yaml") {
+				return nil
+			}
+
+			// Read kustomization file.
+			kustomizationFileBytes, err := os.ReadFile(path) //nolint:gosec
+			if err != nil {
+				return err
+			}
+			kustomizationFileContent := kustomizationFile{}
+			if err := yaml.Unmarshal(kustomizationFileBytes, &kustomizationFileContent); err != nil {
+				return err
+			}
+
+			baseDir := filepath.Dir(path)
+
+			// Iterate through helmCharts in the kustomization.yaml and cleanup outdated charts.
```

Comment:
What if we decide to unpin a chart version? should we skip the cleanup or cleanup everything so we force using latest.
I'm leaning toward the first option given that it provides more stability and there is make clean-charts as escape path; the code however seems to fall in the second option, but I'm not sure this is intentional given that e.g. the log message will look weird if the version is not set in kustomize.yaml

Reply:
I'll take a look at the API used for Helm chart repos. Maybe there's an easy way to figure out what latest is. Then I can resolve latest against the helm repo and delete when necessary with the same logic as of today

Reply:
Now it should also work with latest :)

Tested it locally and I think it works.

This can be reproduced by:
* dropping the version from one of the kustomization.yamls
* changing the version in Chart.yaml to another version
* renaming the tgz file according to the same version

Running tilt up or tilt-prepare

```
I0908 12:51:53.473061  310605 main.go:635] [resources/grafana.observability] resolved latest Helm Chart version to "6.37.0"
I0908 12:51:53.473282  310605 main.go:650] [resources/grafana.observability] local Helm Chart is outdated (6.37.0 != 6.35.0), deleting cached chart...

```